File: Animation-OC/Animation-OC/OpacityViewController.m


//
//  OpacityViewController.m
//  Animation-OC
//
//  Created by chenyufeng on 15/12/31.
//  Copyright © 2015年 chenyufengweb. All rights reserved.
//

#import "OpacityViewController.h"

@interface OpacityViewController ()

@property (weak, nonatomic) IBOutlet UIView *redSquare;

@end

@implementation OpacityViewController

- (void)viewDidLoad {

  [super viewDidLoad];
}

- (void)viewWillAppear:(BOOL)animated{

  [super viewWillAppear:animated];
  [UIView animateWithDuration:2 animations:^{
    self.redSquare.alpha = 0.3;
  }];
}

@end

File: PositionAnimation01/PositionAnimation01/OpacityViewController.swift
//
//  OpacityViewController.swift
//  PositionAnimation01
//
//  Created by chenyufeng on 15/8/20.
//  Copyright (c) 2015年 chenyufengweb. All rights reserved.
//

import UIKit

class OpacityViewController: UIViewController {

    @IBOutlet weak var greenSquare: UIView!
  
    override func viewDidLoad() {

        super.viewDidLoad()
    }
  
    override func viewDidAppear(animated: Bool) {

        super.viewDidAppear(animated)
        func anim(){

            self.greenSquare.alpha = 0.2   //改变透明度到0.2
        }
        UIView.animateWithDuration(2, animations: anim)//时常为2s；
    }
}

